todo_app/app.py

```diff
@@ -1,11 +1,16 @@
-from flask import Flask
-
+from flask import Flask, render_template, request
 from todo_app.flask_config import Config
+from todo_app.data.session_items import get_items, add_item
 
 app = Flask(__name__)
 app.config.from_object(Config())
 
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    return render_template('index.html', items=get_items())
+
+@app.route('/item', methods=['POST'])
+def add_new_item():
+    add_item(request.form.get('name'))
+    return render_template('index.html', items=get_items())
```

Comment:
I'm a fan of using a redirect here (as below, imported from flask) over this direct return for a couple of reasons:
* If your index gets more complicated, then you've got duplicated things here that will need to be updated as well
* If someone refreshes the page after submitting a new item, it will try to create another
* If someone bookmarks the URL after hitting this method, they'll see errors because they'll later be trying to GET this URL

```python
    return redirect('/')
```

It does have a disadvantage that showing error messages needs a bit more engineering (since the error is essentially generated here, but needs to propagate past the refresh) but I think that's a reasonable trade-off